test-utils/src/main/kotlin/Utils.kt

```diff
@@ -94,4 +94,18 @@ fun patchActualResult(
 ): String =
     patchActualResult(yamlMapper.writeValueAsString(result), patchStartAndEndTime)
 
-fun readOrtResult(file: String) = File(file).let { it.mapper().readValue<OrtResult>(patchExpectedResult(it)) }
+fun patchActualResultObject(result: OrtResult, patchStartAndEndTime: Boolean = false): OrtResult =
+    yamlMapper.readValue(patchActualResult(result, patchStartAndEndTime))
+
+fun readOrtResult(file: String) = readOrtResult(File(file))
+
+fun readOrtResult(file: File) = file.mapper().readValue<OrtResult>(patchExpectedResult(file))
+
+/**
+ * Convert a result in serialized form by deserializing it and then serializing it again. This causes a conversion to
+ * the most recent serialization format.
+ */
+fun convertResult(orgResult: String): String {
+    val ortResult = yamlMapper.readValue<OrtResult>(orgResult)
+    return yamlMapper.writeValueAsString(ortResult)
```

Comment:
Without knowing about the custom serializer, it would be totally unclear to the read why this would convert anything. So I wonder whether we should rename this function so something that doe snot include "convert" at all, like "reserializeOrtResult" or so.

Reply:
Also "orgResult" and "ortResult" look awfully similar. "org" could also be misinterpreted as "organization". So why not just "result"? Or "oldResult" and "newResult"?

Reply:
I renamed the function to `applyCustomSerialization()` and gave some more background in the comment.

Reply:
I'm not sure if I understand why this function is required. Couldn't the same be achieved by just fixing the expected result files once to the new format?

Reply:
Yes, this would be possible, but this would have some drawbacks:
* Many files would have to be adapted, as there are a lot of expected result files.
* The new format is less readable.
* When you change both the implementation and the expected results, you cannot really be sure whether the new implementation yields correct results.

Therefore, I would prefer to keep the expectations in the old format, so to say as a reference. The tests then have to do some kind of conversion to check the content of the produced results.

Reply:
In my view the only argument that really counts somewhat is "The new format is less readable". But how readable do expected test result files actually need to be? I still need to make up my mind about this.

Reply:
With the old format, you can understand the dependency relations between projects and packages quite easily. This is not possible with the new format where you just have a tree of indices. So with reasonable effort, it is hardly possible to derive test expectations from the result file.

Reply:
I just realized that my last comment was wrong, in case `applyCustomSerialization` is used the new format would be shown in the diff, right? Anyway, I would be fine with merging this as is and discuss later if we want to change this.